File: examples-sdk/csharp/004-basic-rest-crud/Main.cs
// 004-basic-rest-crud — C# SDK equivalent
// Compile: vil compile --from csharp --input 004-basic-rest-crud/Main.cs --release

#r "sdk/csharp/Vil.cs"

var server = new VilServer("crud-vilorm", 8080);
var tasks = new ServiceProcess("tasks");
tasks.Endpoint("GET", "/tasks", "list_tasks");
tasks.Endpoint("POST", "/tasks", "create_task");
tasks.Endpoint("GET", "/tasks/stats", "task_stats");
tasks.Endpoint("GET", "/tasks/:id", "get_task");
tasks.Endpoint("PUT", "/tasks/:id", "update_task");
tasks.Endpoint("DELETE", "/tasks/:id", "delete_task");
server.Service(tasks);
server.Compile();

File: examples-sdk/kotlin/004-basic-rest-crud/main.kt
// 004-basic-rest-crud — Kotlin SDK equivalent
// Compile: vil compile --from kotlin --input 004-basic-rest-crud/main.kt --release

fun main() {
    val server = VilServer("crud-vilorm", 8080)
    val tasks = ServiceProcess("tasks")
    tasks.endpoint("GET", "/tasks", "list_tasks")
    tasks.endpoint("POST", "/tasks", "create_task")
    tasks.endpoint("GET", "/tasks/stats", "task_stats")
    tasks.endpoint("GET", "/tasks/:id", "get_task")
    tasks.endpoint("PUT", "/tasks/:id", "update_task")
    tasks.endpoint("DELETE", "/tasks/:id", "delete_task")
    server.service(tasks)
    server.compile()
}
